hnn_core/tests/test_network.py

```diff
@@ -226,6 +228,24 @@ def test_network():
             kwargs[arg] = string_arg
             net.add_connection(**kwargs)
 
+    # Check probability=0.5 produces half as many connections as default
+    net.add_connection(**kwargs_default)
+    kwargs = kwargs_default.copy()
+    kwargs['probability'] = 0.5
+    net.add_connection(**kwargs)
+    n_connections = np.sum(
+        [len(t_gids) for
+         t_gids in net.connectivity[-2]['gid_pairs'].values()])
+    n_connections_new = np.sum(
+        [len(t_gids) for
+         t_gids in net.connectivity[-1]['gid_pairs'].values()])
+    assert n_connections_new == np.round(n_connections * 0.5).astype(int)
```

Comment:
Is this safe? Will it always be `==`, or could there be a rounding error (+/- 1)?

Reply:
Good question. Under the hood, I am gathering all connections into a list. I then calculate how many connections should be retained with `n_connections * probability`, and then use `np.random.choice(conn_list, size=n_conn, replace=False)`. This guarantees that there will always be a predictable number of connections. 

Perhaps I can be more clear in the docs, as some people might expect a probabilistic yes/no for every possible connection. In that case there may be slight variation.

Reply:
> This guarantees that there will always be a predictable number of connections.

nice!